presto-main/src/main/java/com/facebook/presto/operator/NestedLoopJoinOperator.java

```diff
@@ -217,59 +224,120 @@ public void close()
         afterClose.run();
     }
 
+    @VisibleForTesting
+    static NestedLoopOutputIterator createNestedLoopOutputIterator(Page probePage, Page buildPage)
```

Comment:
This PR might be structurally cleaner to put this change (the new `PageRepeatingIterator` and `PageRepeatingIterator` logic) in a separate commit other than the change within `NestedLoopPageBuilder` class. Maybe put different things in different commits in the future PRs? It'll be easier to review.

Reply:
Sure, will keep that in mind next time.